moped-database/migrations/1676056649901_dedupe_subcomponents_table/up.sql

```diff
@@ -0,0 +1,54 @@
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 11;
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 12;
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 13;
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 14;
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 15;
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 16;
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 17;
+update public.moped_proj_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 18;
+-- and now we need to do this for our new moped_components_subcomponents table
+update public.moped_components_subcomponents set subcomponent_id = 1 where subcomponent_id = 4;
+update public.moped_components_subcomponents set subcomponent_id = 2 where subcomponent_id = 5;
+update public.moped_components_subcomponents set subcomponent_id = 3 where subcomponent_id = 6;
+update public.moped_components_subcomponents set subcomponent_id = 1 where subcomponent_id = 7;
+update public.moped_components_subcomponents set subcomponent_id = 2 where subcomponent_id = 8;
+update public.moped_components_subcomponents set subcomponent_id = 3 where subcomponent_id = 9;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 10;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 11;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 12;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 13;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 14;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 15;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 16;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 17;
+update public.moped_components_subcomponents set subcomponent_id = 4 where subcomponent_id = 18;
+
+-- ok, now delete all those redundant subcomponents
+DELETE FROM public.moped_subcomponents WHERE subcomponent_id > 4;
+-- and reset the table ID sequence
+SELECT setval('moped_subcomponents_subcomponent_id_seq', 5, TRUE);
+
+-- and add a unique constraint on the name for good measure
+ALTER table public.moped_subcomponents add constraint subcomponent_name_unique UNIQUE (subcomponent_name);
+-- and drop the component_id column, which has moved to moped_components_subcomponents
+-- todo: make metadata match
+ALTER table public.moped_subcomponents drop column component_id;
```

Comment:
😵 
we're going from a table that looks like this:

component_id | subcomponent_id | subcomponent_name
-- | -- | --
16 | 1 | Audible push button'
16 | 2 | Bicycle signal'
16 | 3 | Leading pedestrian interval'
17 | 4 | Audible push button'
17 | 5 | Bicycle signal'
17 | 6 | Leading pedestrian interval'
18 | 7 | Audible push button'
18 | 8 | Bicycle signal'
18 | 9 | Leading pedestrian interval'
7 | 10 | Raised pavement markers'
8 | 11 | Raised pavement markers'
9 | 12 | Raised pavement markers'
10 | 13 | Raised pavement markers'
11 | 14 | Raised pavement markers'
12 | 15 | Raised pavement markers'
6 | 16 | Raised pavement markers'
21 | 17 | Raised pavement markers'
22 | 18 | Raised pavement markers'

to a table that looks like this:


subcomponent_id | subcomponent_name
-- | --
1 | Audible push button'
2 | Bicycle signal'
3 | Leading pedestrian interval'
4 | Raised pavement markers'